src/services/OAUTH2.ts

```diff
@@ -0,0 +1,91 @@
+import { EndpointInfo } from '../types'
+import { generateRandomState, getQueryParameter } from './functions'
+
+export const oauth2Request = async (
+  { short, method }: EndpointInfo,
```

Comment:
You terminated general request and left only oauth 
It's incorrect from architecture perspective because you eliminated universal item in favor less universal 
I propose use oauth like a flag

Reply:
fixed